### PR TITLE
Update glossary.md

### PR DESCRIPTION
**Pull request by wesleywiser**
- Added additional information to some of the definitions.

- Made term capitalization consistent so that only terms which refer to acronyms or concrete types are capitalized. Please let me know if that isn't desirable for some reason.

### CODE REVIEW

File: src/appendix/glossary.md [hunk 1]
```diff
@@ -18,21 +18,21 @@ Term                                           | Meaning
...
 <span id="free-var">free variable</span>       |  A "free variable" is one that is not bound within an expression or term; see [the background chapter for more](./background.md#free-vs-bound)
 <span id="generics">generics</span>            |  The set of generic type parameters defined on a type or item.
 <span id="hir">HIR</span>                      |  The High-level IR, created by lowering and desugaring the AST. ([see more](../hir.md))
 <span id="hir-id">HirId</span>                 |  Identifies a particular node in the HIR by combining a def-id with an "intra-definition offset". See [the HIR chapter for more](../hir.md#identifiers-in-the-hir).
-<span id="hir-map">HIR Map</span>              |  The HIR map, accessible via tcx.hir, allows you to quickly navigate the HIR and convert between various forms of identifiers.
+<span id="hir-map">HIR map</span>              |  The HIR map, accessible via tcx.hir, allows you to quickly navigate the HIR and convert between various forms of identifiers.
```
Comment (jyn514) [inline src/appendix/glossary.md:32]:
```suggestion
<span id="hir-map">HIR map</span>              |  The HIR map, accessible via `tcx.hir()`, allows you to quickly navigate the HIR and convert between various forms of identifiers.
```

File: src/appendix/glossary.md [hunk 2]
```diff
@@ -73,21 +73,21 @@ Term                                           | Meaning
...
 <span id="token">token</span>                  |  The smallest unit of parsing. Tokens are produced after lexing ([see more](../the-parser.md)).
 <span id="tls">[TLS]</span>                    |  Thread-Local Storage. Variables may be defined so that each thread has its own copy (rather than all threads sharing the variable). This has some interactions with LLVM. Not all platforms support TLS.
 <span id="trait-ref">trait reference</span>    |  The name of a trait along with a suitable set of input type/lifetimes. ([see more](../traits/goals-and-clauses.md#trait-ref))
-<span id="trans">trans</span>                  |  The code to translate MIR into LLVM IR. Renamed to codegen.
-<span id="ty">ty</span>                        |  The internal representation of a type. ([see more](../ty.md))
-<span id="tyctxt">TyCtxt</span>                |  The data structure often referred to as [tcx](#tcx) in code
+<span id="trans">trans</span>                  |  Short for "translation", the code to translate MIR into LLVM IR. Renamed to codegen.
+<span id="ty">Ty</span>                        |  The internal representation of a type. ([see more](../ty.md))
```
Comment (LeSeulArtichaut) [inline src/appendix/glossary.md:83]:
```suggestion
<span id="ty">`Ty`</span>                        |  The internal representation of a type. ([see more](../ty.md))
```

**Review by jyn514 — COMMENTED** [review]
Thanks so much!

File: src/appendix/glossary.md [hunk 3]
```diff
@@ -18,21 +18,21 @@ Term                                           | Meaning
...
-<span id="hir-map">HIR Map</span>              |  The HIR map, accessible via tcx.hir, allows you to quickly navigate the HIR and convert between various forms of identifiers.
+<span id="hir-map">HIR map</span>              |  The HIR map, accessible via tcx.hir, allows you to quickly navigate the HIR and convert between various forms of identifiers.
 <span id="ice">ICE</span>                      |  Short for internal compiler error, this is when the compiler crashes.
 <span id="ich">ICH</span>                      |  Short for incremental compilation hash, these are used as fingerprints for things such as HIR and crate metadata, to check if changes have been made. This is useful in incremental compilation to see if part of a crate has changed and should be recompiled.
-<span id="infcx">infcx</span>                  |  The inference context (see `rustc_middle::infer`)
+<span id="infcx">infcx</span>                  |  The type inference context (see `rustc_middle::infer`).
```
Comment (LeSeulArtichaut) [inline src/appendix/glossary.md:35]: I think the name `InferCtxt` could appear here

**Review by LeSeulArtichaut — APPROVED** [review]
Looks good :+1:

I left a few nits which were there before this PR and could be fixed in a follow-up, feel free to ignore.

File: src/appendix/glossary.md (hunk 3 above)
Comment (wesleywiser) [inline src/appendix/glossary.md:35]:
```suggestion
<span id="infcx">infcx</span>                  |  The type inference context (`InferCtxt`). (see `rustc_middle::infer`)
```

**Comment by wesleywiser** [comment]
Suggestions applied!

**Review by LeSeulArtichaut — APPROVED** [review]
Thanks!

**Pull request merged**